pkg/services/rendering/http_mode.go

```diff
@@ -171,20 +174,13 @@ func (rs *RenderingService) doRequest(ctx context.Context, u *url.URL, headers m
 	return resp, nil
 }
 
-func (rs *RenderingService) readFileResponse(ctx context.Context, resp *http.Response, filePath string, url string) error {
+func (rs *RenderingService) writeResponseToFile(ctx context.Context, resp *http.Response, filePath string) error {
 	// check for timeout first
 	if errors.Is(ctx.Err(), context.DeadlineExceeded) {
 		rs.log.Info("Rendering timed out")
 		return ErrTimeout
 	}
 
-	// if we didn't get a 200 response, something went wrong.
```

Comment:
Remove this block. This function shouldn't be called with an invalid resp. This error is not related to the func responsibility.